packages/ui-source-code-editor/src/SourceCodeEditor/README.md

```diff
@@ -0,0 +1,1056 @@
+
+```js
+---
+example: true
+render: false
+background: light
+---
+class FocusExample extends React.Component {
+  state = {
+    indentWithTab: true,
+    indentUnitCount: '2',
+  }
+
+  editor = null
+
+  render () {
+    return (
+      <View display="block" padding="medium medium small" background="primary">
+        <View display="block" margin="small none large">
+          <Button onClick={() => {
+            console.log('manual focus')
+            this.editor.focus()
+          }}>
+            Focus editor
+          </Button>
+        </View>
+
+        <SourceCodeEditor
+          label='focus example'
+          ref={(component) => { this.editor = component }}
+          language="jsx"
+          onFocus={() => {
+            console.log('onFocus')
+            console.log({ hasFocus: this.editor.hasFocus })
+          }}
+          onBlur={() => {
+            console.log('onBlur')
+            console.log({ hasFocus: this.editor.hasFocus })
+          }}
+          defaultValue={`function exampleMethod(props: Props) {
+  return props ? props.value : null
+}`}
+        />
+      </View>
+    )
+  }
+}
+
+render(<FocusExample />)
+```
+
+### Attachment
+
+The `attachment` prop removes the top/bottom border-radius and margin of the editor, so it can be attached to the top or bottom of another element.
+
+```js
+---
+example: true
+render: false
+background: light
+---
+class AttachmentExample extends React.Component {
+  state = {
+    attachment: 'none'
+  }
+
+  render () {
+    return (
+      <View display="block" padding="medium medium small" background="primary">
+        <View display="block" margin="small none large">
+          <RadioInputGroup
+            name="attachmentExample"
+            value={this.state.attachment}
+            description="attachment"
+            onChange={(e, attachment) => {
+              this.setState({attachment})
+            }}
+          >
+            {['none', 'top', 'bottom'].map(attachment => <RadioInput key={attachment} label={attachment} value={attachment} />)}
+          </RadioInputGroup>
+        </View>
+
+        {this.state.attachment === 'bottom' && (
+          <View as="div" borderWidth='small' background='secondary' padding="small">
+            CodeEditor is attached to the bottom of this element
+          </View>
+        )}
+        <SourceCodeEditor
+          label='attachment example'
+          language="jsx"
+          attachment={this.state.attachment === 'none' ? undefined : this.state.attachment}
+          defaultValue={`const fruit: string = "apple"
+
+function exampleMethod(props: Props) {
+  return props ? props.value : null
+}`}
+        />
+        {this.state.attachment === 'top' && (
+          <View as="div" borderWidth='small' background='secondary' padding="small">
+            CodeEditor is attached to the top of this element
+          </View>
+        )}
+      </View>
+    )
+  }
+}
+
+
+
+render(<AttachmentExample />)
+```
+
+### Playground
```

Comment:
Well, I was planning to delete this, because it was just made for building and testing purposes :) The other examples cover all of these options in interactive ways too. I feel like it is redundancy, but do you think we should keep it after all?

Reply:
yep, then its not needed.